### DISCUSSION TITLE
compatibility and average RGB

### DISCUSSION THREAD

**Discussion by bcpi** [SixLabors/ImageSharp#1219] [Q&A] [answered]
Hi folks,

Congrats on creating this library! 

I have a .NET Core 2.1 web api project which we plan on doing some image manipulation. Mainly, we just need to resize JPG and PNG photos uploads, but also extract some information from the photos. One feature we need is obtaining the average RGB value of a photo. Is this something we can accomplish with ImageSharp? Is the library also compatible with .NET core 2.1 under windows?

Thanks!

**Comment by Ryder25** [comment, marked as answer]
The library gives you access to the pixels of the image so you can average them yourself. I'm not sure if there's a built-in function to do it for you.

According to the main Readme.md, the library is built against .Net Standard 1.3 so it will run on .Net Core 2.1.

Note: I'm not a dev for this library.

**Reply by JimBobSquarePants** [reply]
@Ryder25 is correct on both counts.

You'll need write the averaging code yourself but you can get fast access to a pixel row to calculate averages via the following method.

https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Image-1.html#SixLabors_ImageSharp_Image_1_GetPixelRowSpan_System_Int32_

**Reply by bcpi** [reply]
Thank you both for the info, much appreciated! 

Any examples online on how to loop through the pixel rows with GetPixelRowSpan()? 

TIA

**Reply by JimBobSquarePants** [reply]
Happy to help! 

You can loop through the pixels of an image in the following manner.

```c#
using var image = Image<Rgba32>.Load(....);
for (int y = 0; i < image.Height; y++) 
{
    Span<Rgba32> rowSpan = image.GetPixelRowSpan(y);
    for (int x = 0; x < rowSpan.Length; x++) 
    {
         Rgba32 pixel = rowSpan[x];
    }
}
```

**Reply by bcpi** [reply]
Excellent! Thank you so much for your help! :)
